nautobot/docs/user-guide/administration/guides/request-profiling.md

```diff
@@ -0,0 +1,77 @@
+# Request Profiling
+
+Nautobot offers advanced request profiling through [django-silk](https://github.com/jazzband/django-silk). This allows administrators to collect debug information about user activities, which can be used to troubleshoot issues with the system.
+
+## User Setting
+
+Request profiling may be enabled by individual users in their profile within the web interface. This can be found under the "Advanced Settings" section.
+
+![user advanced settings](../../../media/user-guide/administration/guides/request-profiling/advanced-settings.png)
+
+Once a user enables request profiling, all subsequent HTTP requests made by that specific user to the system will be logged by django-silk. This setting will persist until the user either logs out or disables the setting in their profile.
+
+## Silk UI
+
+Nautobot administrators with super-user permissions can access the django-silk UI at the `/silk/` URL.
+
+![silk ui](../../../media/user-guide/administration/guides/request-profiling/silk-ui.png)
+
+From there, administrators can view details of individual requests, including timing, SQL queries, and cProfile artifacts.
+
+## Configuration
+
+The Nautobot configuration comes out of the box with django-silk set up to support the above functionality. Those settings are described below, but it is not the intention of the Nautobot docs to describe all django-silk settings. Django-silk provides several other parameters that knowledgeable users may also be able to use, depending on the use case.
+
+### SILKY_PYTHON_PROFILER
+
+Default: `True`
```

Comment:
Every profiler i know provides a data load on it, I would have thought we would default to false and add turn on when in "DEBUG" mode.

Reply:
Not sure what you mean by "data load." This setting doesn't actually enable or turn anything on. Requests are only profiled if the user turns the setting on.

Reply:
Sorry, was multi-tasking... I just meant adds load (CPU/Memory/IO) to the server.

>  Requests are only profiled if the user turns the setting on.

This "Enables use of the built-in Python cProfile profiler." seems to me to imply it would be the thing that turns the setting on.

Reply:
I can see why you might think that :)

It is true that silk is technically always "on", but because of the intercept function, every request is immediately short circuited _unless_ the setting is enabled on the user's session.

`SILKY_PYTHON_PROFILER = True` is just saying, when silk does actually intercept a request, it should use cProfile.

Reply:
I'll add, the point of this setup is to allow quick, reactive profiling, without the need to reconfigure the environment. I.e. I can do it in a production environment. Because it is always scoped to a user, the performance impact is not a concern. Silk even lets you quantify its own impact in overall profile.

This is a pretty common implementation pattern.